### ISSUE TITLE
Support loading a project directly from s3

### ISSUE THREAD

**Issue by constantinpape** [mobie/mobie-viewer-fiji#182]
Enable opening projects from s3 directly, without the need for github.
This would make it easier to share data for users who are not familiar with git/github.

Having the metadata on github should always stay an option for legacy with old projects and to have version control for the metadata.

**Comment by constantinpape** [comment]
@tischi I can maybe look into this, but I would need some pointers to the relevant parts in the code.

**Comment by constantinpape** [comment]
@tischi quick reminder to point me to the relevant part of the viewer code.

**Comment by tischi** [comment]
@constantinpape

https://github.com/mobie/mobie-viewer-fiji/blob/1c4f8d296fe505f4f421ef101b428ba222db0480/src/main/java/de/embl/cba/mobie/ui/MoBIE.java#L60

**Comment by constantinpape** [comment]
I have implemented access for public buckets in #211.

Now I am looking into implementing access from private buckets (i.e. buckets that need credentials for read access).
While working on this I noticed a few things and ran into some issues:
1. It's actually possible to access files in public buckets via HTTPS (I initially thought this wasn't possible), but we still need special treatment for s3 in order to list objects etc.
2. It's not necessary to know beforehand if a bucket is public or private, so we could get rid of the "Protected" and "Anonymous" fields in the xml. I don't actually know how to do this in java yet, only in python, I will have a look into how to do it in java and then refactor the code a bit.
3. Accessing files in private buckets via HTTPS doesn't work. This currently prevents us from opening a project from a private buket, because the bdv-xmls are still streamed via HTTPS. I am not quite sure what's the best way to fix this, because it looks to me like this is happening in `mpicbg.spim.data.generic.XmlIoAbstractSpimData.load` so not in a place we can easily change. See full stacktrace below for details.

@tischi I think it's best if we discuss this via zoom or phone (I have time all Monday).

For reference, here is the full traceback when trying to open from a private bucket via HTTPS.
```
Adding source: em-raw...
https://s3.embl.de/platybrowser-private/example-project/example-dataset/images/remote/em-raw.xml
mpicbg.spim.data.SpimDataIOException: java.io.IOException: Server returned HTTP response code: 403 for URL: https://s3.embl.de/platybrowser-private/example-project/example-dataset/images/remote/em-raw.xml
	at mpicbg.spim.data.generic.XmlIoAbstractSpimData.load(XmlIoAbstractSpimData.java:88)
	at de.embl.cba.bdv.utils.BdvUtils.openSpimData(BdvUtils.java:1274)
	at de.embl.cba.bdv.utils.sources.LazySpimSource.initSpimData(LazySpimSource.java:89)
	at de.embl.cba.bdv.utils.sources.LazySpimSource.wrappedVolatileSource(LazySpimSource.java:66)
	at de.embl.cba.bdv.utils.sources.LazySpimSource.getType(LazySpimSource.java:137)
	at de.embl.cba.bdv.utils.sources.LazySpimSource.getType(LazySpimSource.java:48)
	at bdv.util.BdvFunctions.addSource(BdvFunctions.java:673)
	at bdv.util.BdvFunctions.show(BdvFunctions.java:224)
	at de.embl.cba.mobie.ui.SourcesDisplayManager.addSourceToBDV(SourcesDisplayManager.java:461)
	at de.embl.cba.mobie.ui.SourcesDisplayManager.showIntensitySource(SourcesDisplayManager.java:444)
	at de.embl.cba.mobie.ui.SourcesDisplayManager.show(SourcesDisplayManager.java:415)
	at de.embl.cba.mobie.bookmark.BookmarkManager.show(BookmarkManager.java:93)
	at de.embl.cba.mobie.bookmark.BookmarkManager.setView(BookmarkManager.java:58)
	at de.embl.cba.mobie.ui.MoBIE.lambda$new$0(MoBIE.java:81)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at org.GNOME.Accessibility.AtkWrapper$6.dispatchEvent(AtkWrapper.java:705)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Caused by: java.io.IOException: Server returned HTTP response code: 403 for URL: https://s3.embl.de/platybrowser-private/example-project/example-dataset/images/remote/em-raw.xml
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1900)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1498)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:268)
	at org.apache.xerces.impl.XMLEntityManager.setupCurrentEntity(Unknown Source)
	at org.apache.xerces.impl.XMLVersionDetector.determineDocVersion(Unknown Source)
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
	at org.apache.xerces.parsers.XMLParser.parse(Unknown Source)
	at org.apache.xerces.parsers.AbstractSAXParser.parse(Unknown Source)
	at org.apache.xerces.jaxp.SAXParserImpl$JAXPSAXParser.parse(Unknown Source)
	at org.jdom2.input.sax.SAXBuilderEngine.build(SAXBuilderEngine.java:217)
	at org.jdom2.input.sax.SAXBuilderEngine.build(SAXBuilderEngine.java:329)
	at org.jdom2.input.SAXBuilder.build(SAXBuilder.java:1266)
	at mpicbg.spim.data.generic.XmlIoAbstractSpimData.load(XmlIoAbstractSpimData.java:84)
	... 28 more
Exception in thread "AWT-EventQueue-1" java.lang.NullPointerException
	at bdv.BigDataViewer.initSetups(BigDataViewer.java:293)
	at de.embl.cba.bdv.utils.sources.LazySpimSource.initSpimData(LazySpimSource.java:92)
	at de.embl.cba.bdv.utils.sources.LazySpimSource.wrappedVolatileSource(LazySpimSource.java:66)
	at de.embl.cba.bdv.utils.sources.LazySpimSource.getType(LazySpimSource.java:137)
	at de.embl.cba.bdv.utils.sources.LazySpimSource.getType(LazySpimSource.java:48)
	at bdv.util.BdvFunctions.addSource(BdvFunctions.java:673)
	at bdv.util.BdvFunctions.show(BdvFunctions.java:224)
	at de.embl.cba.mobie.ui.SourcesDisplayManager.addSourceToBDV(SourcesDisplayManager.java:461)
	at de.embl.cba.mobie.ui.SourcesDisplayManager.showIntensitySource(SourcesDisplayManager.java:444)
	at de.embl.cba.mobie.ui.SourcesDisplayManager.show(SourcesDisplayManager.java:415)
	at de.embl.cba.mobie.bookmark.BookmarkManager.show(BookmarkManager.java:93)
	at de.embl.cba.mobie.bookmark.BookmarkManager.setView(BookmarkManager.java:58)
	at de.embl.cba.mobie.ui.MoBIE.lambda$new$0(MoBIE.java:81)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at org.GNOME.Accessibility.AtkWrapper$6.dispatchEvent(AtkWrapper.java:705)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

**Comment by constantinpape** [comment]
Loading from public and private s3 buckets directly works thanks to #215. This has two advantages:
- It's easier for users to share their projects, because they don't need to put anything on github.
- Data can be kept completely private by pushing everything to a private bucket.

 We should mention this in the corresponding parts of the tutorials etc. now. Cc @K-Meech.
